README.md

```diff
@@ -28,8 +29,52 @@ Optionally, contributions can include:
 * A test suite following the testing framework in our `-ts` templates.
 
 
-To submit a contribution, [open a pull request](https://github.com/jspsych/jspsych-contrib/pulls).
-In the pull request, please make it clear how we can verify that the contribution is functional. 
+To submit a contribution, [open a pull request](https://github.com/jspsych/jspsych-contrib/pulls) that contains a directory for your plugin/extension inside the `/packages` directory.
+In the pull request comments, please make it clear how we can verify that the contribution is functional. 
 This could be accomplished with a link to a demonstration experiment, the inclusion of an example file and/or testing files, or through some other means.
 We try to review pull requests quickly and add new contributions as soon as the minimal standards are met.
 
+## Plugin templates
+
+There are two plugin template directories inside the `/packages` directory that you can use as a reference when creating a directory for your plugin contribution. 
+Both templates are compatible with jsPsych v7+.
+
+Regardless of which template you use, you can get started by creating a copy of the template folder in `/packages` and renaming it according to your plugin/extension name. Be sure that you also:
+* Edit the `package.json` file
+* Add a readme.md file to your plugin/extension directory, based on the [readme template](readme-template.md)
+
+### `plugin-template-ts`
+
+This template uses TypeScript source files that are complied into JavaScript using Node.js and npm.
+This is the format used for plugins in the main jsPsych repo.
+To use this template, you should edit the `src/index.ts` file, keeping the overall structure but changing the details as appropriate (plugin name, parameters, trial method, etc.).
+You can then use the `npm run build` command to compile your `index.ts` code into JavaScript files, which will appear in a `/dist` directory.
+This format also allows you to add a Jest test file (optional), which you can create based on the `src/index.spec.ts` template file.
+
+In the `rollup.config.mjs` file, replace "jsPsychPluginName" with your plugin name (same as the class name in the `index.ts` file).
+You do not need to edit the other config files in this template directory.
```

Comment:
Is this right?

Reply:
I find "same as the class name in the `index.ts` file" a bit confusing, since `jsPsychPluginName` != `PluginName`...

Reply:
Oh right, oops. I'll remove that. Is it correct to say this?
"In the `rollup.config.mjs` file, replace "jsPsychPluginName" with your plugin name. This will be the name that can be used as the value of the `type` parameter when creating a trial with your plugin."

Reply:
Done in f1478ee

Reply:
> Is it correct to say this?

Sorry, didn't see this! Absolutely right!